server/task/src/main/java/org/apache/james/task/TaskManager.java

```diff
@@ -58,6 +60,62 @@ public boolean isFinished() {
         }
     }
 
+    interface AwaitedTaskExecutionDetails {
```

Comment:
Isn't this a disguised sketchy staged builder?

Reply:
A staged builder of one element? it will only make it heavier for no benefit

Reply:
Well I have the impression that you return several time invalid variations of the same entity, for which calling unwrap fails with a not specific exception.

You require IMO too much knowledge from the caller (if I use it, I will break it) and I fail to understand the way this "thing" - that do not follow patterns that I know - work.

Can we try to review and rediscuss code organization here?

Reply:
No, and I won't fix it.

1. You'll loose in expressiveness (have a look in at tests)
2. You'll add burden to code which is already complex adding no value
3. You have to test the code you produce (including using this one), so you'll be corrected quickly
4. It's perfectly acceptable to have code not behaving the way we are use to, that does not make it a bad code (rephrasing: it's not because we know things that work, that there is nothing we do not know that work too)